### ISSUE TITLE
Mistake in your derivation of the loss objective

### ISSUE THREAD

**Issue by Kyridiculous2** [Mingpan/generative_map#1]
Check out your Equation (2) in the paper:
Expanding line3 with contains KL divergence and expectations doesn't give the original form, another factor is included. See below:
![ddd](https://user-images.githubusercontent.com/26832000/56196846-35632280-606a-11e9-86d5-4606f9d19b32.png)

**Comment by Mingpan** [comment]
Thanks for the feedback!

Actually this is something we intended and not a miss, for the reason that we assume "z should contain all information about p", therefore p( I | z ) = p( I | p, z ), and the additional factor you show in your derivation cancels out. 
This is mentioned (maybe not so explicitly though) in the last sentence of the header of section 3, right before the "3.1 Framework“ subsection, by saying that the generative process goes as p -> z -> I, in a single path.

Maybe a few more words regarding this: The reason for making the assumption above is that we want to infer p (position) simply from looking at z.
If we allow image generation p(I | ... ) to condition on both p and z, we cannot ensure that. p might contain additional information that is not incorporated in z, which is not desired. Because the additional information can prevent p to be inferred by z.
We have a few words in the paper explaining this, in the second paragraph of "VAE-based training objective" of the related work section, where we discuss the difference between our stuff and CVAE.
(I found a typo in that paragraph just now, will fix that immediately)
